src/ipc/ipc-helper.c

```diff
@@ -281,6 +281,10 @@ int ipc_comp_free(struct ipc *ipc, uint32_t comp_id)
 	if (!icd)
 		return -ENODEV;
 
+	/* check type */
+	if (icd->type != COMP_TYPE_COMPONENT)
```

Comment:
can we add a explicit error log to help debugging ?

Reply:
done